arcgis-ios-sdk-samples/Content Display Logic/Controllers/SampleInfoViewController.swift

```diff
@@ -32,24 +32,26 @@ class SampleInfoViewController: UIViewController {
         view.addSubview(webView)
         self.webView = webView
         
-        if self.folderName != nil {
-            self.fetchFileContent(for: self.folderName)
+        if let readmeURL = readmeURL,
+            let html = markdownTextFromFile(at: readmeURL) {
+            displayHTML(html)
         }
     }
     
-    func fetchFileContent(for folderName:String) {
+    func markdownTextFromFile(at url:URL)->String? {
 
-        if let path = Bundle.main.path(forResource: "README", ofType: "md", inDirectory: folderName) {
-            //read the content of the file
-            if let content = try? String(contentsOfFile: path, encoding: String.Encoding.utf8) {
-                //remove the images
-                let pattern = "!\\[.*\\]\\(.*\\)"
-                if let regex = try? NSRegularExpression(pattern: pattern, options: NSRegularExpression.Options.caseInsensitive) {
-                    let newContent = regex.stringByReplacingMatches(in: content, options: NSRegularExpression.MatchingOptions(), range: NSMakeRange(0, content.count), withTemplate: "")
-                    self.displayHTML(newContent)
-                }
+        //read the content of the file
+        if let content = try? String(contentsOf: url, encoding: .utf8) {
+            //remove the images
+            let pattern = "!\\[.*\\]\\(.*\\)"
+            if let regex = try? NSRegularExpression(pattern: pattern, options: NSRegularExpression.Options.caseInsensitive) {
+                let newContent = regex.stringByReplacingMatches(in: content,
+                                                                options: NSRegularExpression.MatchingOptions(),
```

Comment:
Having this much leading whitespace is fairly commonplace in Objective-C, but it feels out of place in Swift. You might consider putting each parameter (including the first one) on a separate line so they aren't indented as much:
```swift
let newContent = regex.stringByReplacingMatches(
    in: content,
    options: NSRegularExpression.MatchingOptions(),
    range: NSMakeRange(0, content.count), withTemplate: ""
)
```
That being said, `NSRegularExpression.MatchingOptions()` is equivalent to the default value of `[]`, so the whole thing could be simplified to:
```swift
let range = NSRange(location: 0, length: content.count)
let newContent = regex.stringByReplacingMatches(in: content, range: range, withTemplate: "")
```

Reply:
Thanks, I'll try it out tomorrow.